internal/config/options.go

```diff
@@ -70,8 +70,8 @@ type Options struct {
 
 	// AuthenticateURL represents the externally accessible http endpoints
 	// used for authentication requests and callbacks
-	AuthenticateURLString string `mapstructure:"authenticate_service_url"`
-	AuthenticateURL       *url.URL
+	AuthenticateURLString string   `mapstructure:"authenticate_service_url"`
+	AuthenticateURL       *url.URL `hash:"ignore"`
```

Comment:
Going to open an issue on this.  The workaround here is fine but I believe something is mutating this at startup.  Unclear if that's expected.

Reply:
Could it be the empty initialization? https://github.com/pomerium/pomerium/blob/1a76a3ad8601aa40aed905204c47c0ee573a6760/internal/config/options.go#L156

Reply:
I don't think so.  It only impacts one of the two URLs and it only happens the first pass through services at start.  I'll follow up researching more thoroughly later that so we can get this merged.